### PR TITLE
Fix printing of exterior powers of modules

### PR DESCRIPTION
**Pull request by HechtiDerLachs**
This should address #3257.

CC: @jankoboehm .

### CODE REVIEW

File: test/Modules/ExteriorPowers.jl [hunk 1]
```diff
@@ -130,12 +130,14 @@ end
   u = (F[1], F[4], F[3])
 
   @test "$(mm(v))" == "e[1]∧e[3]∧e[4]"
-  @test "$(F3)" == "⋀^3(Free module of rank 5 over Multivariate polynomial ring in 5 variables over QQ)"
+  #@test "$(F3)" == "⋀^3(Free module of rank 5 over Multivariate polynomial ring in 5 variables over QQ)"
+  @test "$(F3)" == "3-th exterior power of Free module of rank 5 over Multivariate polynomial ring in 5 variables over QQ"
```
Comment (fingolfin) [inline test/Modules/ExteriorPowers.jl]: should of course be `3rd` not `3-th`.

We have `ordinal_number_string` for this.

Comment (HechtiDerLachs) [reply test/Modules/ExteriorPowers.jl]: Thanks. Didn't know that.

**Comment by lgoettgens** [comment]
This approach has exactly the drawbacks I pointed out in https://github.com/oscar-system/Oscar.jl/issues/3257#issuecomment-1914548021, i.e. the following is not conformant with the unicode preference:
```julia
Oscar.allow_unicode(true)
R, (x,y) = QQ["x","y"]
F = free_module(R, 3)
E = exterior_power(F, 2)
Oscar.allow_unicode(false)
println(gens(E))
```

**Comment by HechtiDerLachs** [comment]
@lgoettgens : You are right. But do we really expect a user to complain about that ever? Who gets the idea to switch on unicode all of a sudden in the middle of some computation?

**Comment by lgoettgens** [comment]
I moved the discussion back to https://github.com/oscar-system/Oscar.jl/issues/3257 with a proposal to adapt the documentation of `allow_unicode`

**Comment by HechtiDerLachs** [comment]
Given @fingolfin 's reply on #3257, this is now ready for merge.

**Review by lgoettgens — APPROVED** [review]

File: test/Modules/ExteriorPowers.jl [hunk 2]
```diff
@@ -129,13 +129,15 @@ end
   v = (F[1], F[3], F[4])
   u = (F[1], F[4], F[3])
 
-  @test "$(mm(v))" == "e[1]∧e[3]∧e[4]"
-  @test "$(F3)" == "⋀^3(Free module of rank 5 over Multivariate polynomial ring in 5 variables over QQ)"
+  @test "$(mm(v))" == "e[1]^e[3]^e[4]"
+  #@test "$(F3)" == "⋀^3(Free module of rank 5 over Multivariate polynomial ring in 5 variables over QQ)"
+  @test "$(F3)" == "3rd exterior power of Free module of rank 5 over Multivariate polynomial ring in 5 variables over QQ"
```
Comment (simonbrandhorst) [inline test/Modules/ExteriorPowers.jl:134]: Free and Multivariate should be lowercase. See the comment above on how to achieve this. 
(I suppose the Multivariate still comes from old module printing). 
In any case this need not block merging of this. It is an improvement in any case.

File: src/Modules/ExteriorPowers/Generic.jl [hunk 3]
```diff
@@ -18,14 +18,22 @@ end
 function show_exterior_product(io::IO, M::ModuleFP)
   success, F, p = is_exterior_power(M)
   success || error("module is not an exterior power")
-  print(io, "⋀^$p($F)")
+  if is_unicode_allowed()
+    print(io, "⋀^$p($F)")
+  else
+    print(io, "$(ordinal_number_string(p)) exterior power of $F")
+  end
```
Comment (simonbrandhorst) [inline src/Modules/ExteriorPowers/Generic.jl:25]:
```suggestion
  io = pretty(io)
  if is_unicode_allowed()
    print(io, "⋀^$p($F)")
  else
    print(io, "$(ordinal_number_string(p)) exterior power of ", Lowercase(), F)
  end
```

Comment (lgoettgens) [reply src/Modules/ExteriorPowers/Generic.jl:25]: bump @HechtiDerLachs

**Comment by simonbrandhorst** [comment]
I merge at this point since this is certainly an improvement.

**Pull request merged**